swaylock/main.c

```diff
@@ -381,59 +381,213 @@ static void load_image(char *arg, struct swaylock_state *state) {
+		{"inside-color", required_argument, NULL, LO_INSIDE_COLOR},
+		{"inside-clear-color", required_argument, NULL, LO_INSIDE_CLEAR_COLOR},
+		{"inside-ver-color", required_argument, NULL, LO_INSIDE_VER_COLOR},
+		{"inside-wrong-color", required_argument, NULL, LO_INSIDE_WRONG_COLOR},
+		{"key-hl-color", required_argument, NULL, LO_KEY_HL_COLOR},
+		{"line-color", required_argument, NULL, LO_LINE_COLOR},
+		{"line-clear-color", required_argument, NULL, LO_LINE_CLEAR_COLOR},
+		{"line-ver-color", required_argument, NULL, LO_LINE_VER_COLOR},
+		{"line-wrong-color", required_argument, NULL, LO_LINE_WRONG_COLOR},
+		{"ring-color", required_argument, NULL, LO_RING_COLOR},
+		{"ring-clear-color", required_argument, NULL, LO_RING_CLEAR_COLOR},
+		{"ring-ver-color", required_argument, NULL, LO_RING_VER_COLOR},
+		{"ring-wrong-color", required_argument, NULL, LO_RING_WRONG_COLOR},
+		{"separator-color", required_argument, NULL, LO_SEP_COLOR},
+		{"text-color", required_argument, NULL, LO_TEXT_COLOR},
+		{"text-clear-color", required_argument, NULL, LO_TEXT_CLEAR_COLOR},
+		{"text-ver-color", required_argument, NULL, LO_TEXT_VER_COLOR},
+		{"text-wrong-color", required_argument, NULL, LO_TEXT_WRONG_COLOR},
 		{0, 0, 0, 0}
 	};
 
 	const char usage[] =
 		"Usage: swaylock [options...]\n"
 		"\n"
+		"  -c, --color <color>            Turn the screen into the given color"
+		" instead of white.\n"
+		"  -e, --ignore-empty-password    When an empty password is provided"
+		" by the user, do not validate it.\n"
+		"  -f, --daemonize                Detach from the controlling terminal"
+		" after locking.\n"
 		"  -h, --help                     Show help message and quit.\n"
-		"  -c, --color <rrggbb[aa]>       Turn the screen into the given color instead of white.\n"
-		"  -s, --scaling                  Scaling mode: stretch, fill, fit, center, tile.\n"
-		"  -t, --tiling                   Same as --scaling=tile.\n"
-		"  -v, --version                  Show the version number and quit.\n"
 		"  -i, --image [<output>:]<path>  Display the given image.\n"
+		"  -s, --scaling <mode>           Scaling mode: stretch, fill, fit,"
+		" center, tile.\n"
+		"  -t, --tiling                   Same as --scaling=tile.\n"
 		"  -u, --no-unlock-indicator      Disable the unlock indicator.\n"
-		"  -f, --daemonize                Detach from the controlling terminal after locking.\n";
+		"  -v, --version                  Show the version number and quit.\n"
+		"  --bs-hl-color <color>          Sets the color of backspace"
+		" highlight segments.\n"
+		"  --font <font>                  Sets the font of the text.\n"
+		"  --indicator-radius <radius>    Sets the indicator radius.\n"
+		"  --indicator-thickness <thick>  Sets the indicator thickness.\n"
+		"  --inside-color <color>         Sets the color of the inside of the"
+		" indicator.\n"
+		"  --inside-clear-color <color>   Sets the color of the inside of the"
+		" indicator when cleared.\n"
```

Comment:
Question for @SirCmpwn: Would it be acceptable to break the 80 character rule for these descriptions? I think it would improve readability, and it is only text. If you're reading the code in an 80 column terminal then you're not missing out on much by having the descriptions go off-screen.

Reply:
Why not just something like

```c
        " --inside-clear-color <color>   "
            "Sets the color of the inside of the indicator when cleared.\n"
```

Reply:
Updated to use @SirCmpwn's suggested format